01-overview-of-ml-systems.md

```diff
@@ -0,0 +1,110 @@
+
+
+4. **Unseen data shares patterns with the train data.**
+
+- Your model only works well if the patterns in the production data are the same than the ones in the training data.
+- *If the data is unseen, how do we know what distribution it comes from?* We don't, we just make a reasonable assumption that it is. We can retrospectively check if our assumption was true by monitoring in production (chapter 8) and testing in production (chapter 9).
+
+
+
+## Problem characteristics that will make ML solutions especially useful
+These are not necessary characteristics, but having one or multiple of them make it much more likely that an ML approach will succeed for a problem. 
+
+1. **It is a repetitive task:**  if a task is repetitive, patterns will be repeated multiple times and the ML models will be able to learn better.
+2. **The cost of wrong predictions is cheap**: ML models will make wrong predictions. When mistakes are cheap, stakes are lower, making it easier to build and deploy ML models. Models can still be built when the consequences of a wrong prediction are catastrophic if on average the benefits of correct predictions outweigh the costs of the wrong ones. Self-driving cars are a good example for this when considering that automated cars on average are safer than cars driven by humans.
+3. **The task happens at scale:** deploying ML models to production is hard. This work is more justifiable when the model will be used to make lots of predictions. Also, if the task happens at scale, there will be lots of data to train from.
+4. **The patterns are constantly changing:** ML shines against traditional software rules when patterns change frequently, causing hardcoded rules get outdated too fast.  Changes in spam trends and culture trends are examples of changing patterns.
+
+
+
+## Typical ML use cases
+ML is being used both for **consumer** facing applications and **enterprise** internal applications. Consumer ML applications is growing; however, as of 2022, the vast majority of applications is still in the enterprise space.
+
+- In **consumer** applications latency tends to be more important than accuracy. Increases in latency may cause the user to bounce from our app. Consumers also tend to be more forgiving with bad predictions in non-critical applications like recommending a movie or bad tagging of a photo.
+- In **enterprise** applications, accuracy tends to be more important than latency. Small increases in accuracy can save millions of dollars per year.  The image below shows the breakdown of ML enterprise applications. These are some examples:
+  - Use ML to reduce cost of customer acquisition (CAC): show better targeted ads, show discounts at the right moment. On  2019, the average CAC for a user that makes an in-app purchase is US $86 61. High CAC is a startup killer.
```

Comment:
To reduce CAC as a service provider, you have to have 1st party data. Like a bank knows you have lots of cash and gives you a discount on opening a self directed trading account. Without 1st party data, you are at the mercy of generic ad platform like FB or Google.